packages/dataviews/src/dataform-controls/integer.tsx

```diff
@@ -81,10 +85,13 @@ export default function Integer< Item >( {
 	const { id, label, description } = field;
 	const value = field.getValue( { item: data } ) ?? '';
 	const onChangeControl = useCallback(
-		( newValue: string | undefined ) =>
+		( newValue: string | undefined ) => {
 			onChange( {
-				[ id ]: Number( newValue ),
-			} ),
+				[ id ]: [ undefined, '', null ].includes( newValue )
```

Comment:
We were converting empty values to 0.

There's an additional bug here that I haven't fixed in this PR. We are assuming the new value will be stored under **data.id**, but it doesn't need to be that way, even if that's the most common scenario. We have `getValue` precisely to take the data from any part of the object (e.g., the field wich `title` id can take its value from `item.title.rendered`). We need a `setValue` equivalent as well. This can be a follow-up PR.

Reply:
Alternatively, we can update `onChange` to only rely the updated value, not the whole item.